### PR TITLE
Make Kotlin-DSL precompiled scripts support strict by default

### PR DESCRIPTION
**Pull request by eskatos**
and deprecate the system property

See 
* https://github.com/gradle/gradle/issues/20416
* https://github.com/gradle/gradle/pull/21029

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 1]
```diff
@@ -108,6 +108,13 @@ Previously Gradle used to detect if an input file collection annotated with `@Sk
 To ignore directories in Gradle 8.0 and later, the input property needs to be explicitly annotated with `@IgnoreEmptyDirectories`.
 For more information see <<empty_directories_file_tree>>.
 
+[[strict-kotlin-dsl-precompiled-scripts-accessors-by-default]]
+=== Strict Kotlin DSL precompiled script plugins accessors generation by default
+
+Type safe Kotlin DSL accessors generation for precompiled script plugins now does fail the build by default if a plugin requested in such precompiled scripts fails to be applied.
+
+The `org.gradle.kotlin.dsl.precompiled.accessors.strict` system property is now deprecated and its usage should be removed. You can find more information <<upgrading_version_7.adoc#strict-kotlin-dsl-precompiled-scripts-accessors, below>> about the introduction of this system property in Gradle 7.6.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:116]:
```suggestion
=== Precompiled script plugins use strict Kotlin DSL accessor generation by default

In precompiled script plugins, type safe Kotlin DSL accessor generation now fails the build if a plugin fails to apply.

Starting in Gradle 7.6, builds could enable this behavior with the `org.gradle.kotlin.dsl.precompiled.accessors.strict` system property. This behavior is now default. The property has been deprecated and its usage should be removed. You can find more information about this property <<upgrading_version_7.adoc#strict-kotlin-dsl-precompiled-scripts-accessors, below>>.
```

Suggestion: simplify some sentences and the section title to make it easier to understand what this feature is and in which situations it applies.

**Review by nathan-contino — APPROVED** [review]
Docs LGTM, one suggestion

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc (hunk 1 above)
Comment (eskatos) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc]: Thank you Nate!

**Review by jbartok — APPROVED** [review]

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Pull request merged**